File: algorithm/typescript/BOJ11866.ts
// problem : https://www.acmicpc.net/problem/11866

import fs = require("fs");

const input: number[] = fs.readFileSync("/dev/stdin").toString().split(" ").map(Number);

const N: number = input[0];
const K: number = input[1];

const array: number[] = new Array(N).fill(0);
const answer: number[] = new Array();

for (let i = 0; i < N; i++) {
  array[i] = i + 1;
}

while (array.length != 0) {
  for (let i = 1; i < K; i++) {
    const num: number = array.shift();
    array.push(num);
  }
  const josephusNum: number = array.shift();
  answer.push(josephusNum);
}

console.log(`<${answer.join(", ")}>`);

File: algorithm/python/BOJ11866.py
# problem : https://www.acmicpc.net/problem/11866

import sys
from collections import deque

N, K = map(int, sys.stdin.readline().rstrip().split())
deq = deque()
for i in range(1, N + 1):
    deq.append(i)

answer = list()

while len(deq) != 0:
    for i in range(1, K):
        num = deq.popleft()
        deq.append(num)
    josephus_num = deq.popleft()
    answer.append(josephus_num)

print("<", end='')
for i in range(0, len(answer) - 1):
    print("{}".format(answer[i]), end=", ")
print("{}>".format(answer[len(answer) - 1]))
